From dbadf67cfe1623012050b14bb547226e69d30682 Mon Sep 17 00:00:00 2001
From: xiangzixuan <xiangzixuan@huawei.com>
Date: Tue, 24 Mar 2026 14:11:08 +0800
Subject: [PATCH 945/960] downstream: iommu: Revert to using .domain_alloc_user
 without kvm parameter

Commit e593cd49 introduced a kvm parameter to arm_smmu_domain_alloc_user()
and switched to .domain_alloc_user_v2. Align the driver by removing
the unused kvm parameter and using .domain_alloc_user again.
No functional change.

Signed-off-by: xiangzixuan <xiangzixuan@huawei.com>
Signed-off-by: liwei <liwei818@huawei.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
index 55635f39d488..7f6676bb6928 100644
--- a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
+++ b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
@@ -3363,7 +3363,6 @@ static struct iommu_domain arm_smmu_blocked_domain = {
 static struct iommu_domain *
 arm_smmu_domain_alloc_user(struct device *dev, u32 flags,
 			   struct iommu_domain *parent,
-			   struct kvm *kvm,
 			   const struct iommu_user_data *user_data)
 {
 	struct arm_smmu_master *master = dev_iommu_priv_get(dev);
@@ -4012,7 +4011,7 @@ static struct iommu_ops arm_smmu_ops = {
 	.hw_info		= arm_smmu_hw_info,
 	.domain_alloc_paging    = arm_smmu_domain_alloc_paging,
 	.domain_alloc_sva       = arm_smmu_sva_domain_alloc,
-	.domain_alloc_user_v2	= arm_smmu_domain_alloc_user,
+	.domain_alloc_user	= arm_smmu_domain_alloc_user,
 	.probe_device		= arm_smmu_probe_device,
 	.release_device		= arm_smmu_release_device,
 	.device_group		= arm_smmu_device_group,
-- 
2.43.0

